### DISCUSSION TITLE
OE3.6version - TerrainConstraintLayer sometimes  has a jagged effect

### DISCUSSION THREAD

**Discussion by iamzheng00** [pelicanmapping/osgearth#2584] [Support Q&A] [answered]
TerrainConstraintLayer worked correctly in OE3.4.These days I tried 3.6version of osgearth and found the effect of `TerrainConstraintLayer `  often has  jagged edges.  Has anyone been in the same situation as me? Or is it a known bug? Or does it need additional configuration options ? 
Here is the code I added TerrainConstraintLayer :
```
osgEarth::OGRFeatureSource* features = new osgEarth::OGRFeatureSource();
// make a geometry:
osgEarth::Polygon* line = new osgEarth::Polygon();
for each (osg::Vec3d vec in pts)
{
	line->push_back(vec);
}
features->setGeometry(line);
features->setProfile(Profile::create(Profile::GLOBAL_GEODETIC));
osgEarth::TerrainConstraintLayer* maskLayer = new osgEarth::TerrainConstraintLayer();
maskLayer->setFeatureSource(features);
maskLayer->setHasElevation(false);
maskLayer->setRemoveInterior(true);
map->addLayer(maskLayer);
```
The result  picture is shown below.
![qqq2](https://github.com/user-attachments/assets/7fb45d5a-9a55-4d57-ae6d-49cab8df13c4)
![qqq](https://github.com/user-attachments/assets/ea1e8661-307d-426d-9370-63655728068f)
![qqq3](https://github.com/user-attachments/assets/a82059b3-1725-4bfe-8c66-24dbbd39b504)

**Comment by gwaldron** [comment, marked as answer]
This should be fixed in f02c28bd7. Thank you.

**Reply by iamzheng00** [reply]
Thank you!
